### ISSUE TITLE
Editor: not possible to publish a page under a non-page menu item

### ISSUE THREAD

**Issue by rralian** [Automattic/wp-calypso#5271]
You can add a number of types of items to menus, like links, pages, posts, etc. But when you are create a new page, the only parent you can give it is a page.

![page-parent](https://cloud.githubusercontent.com/assets/66797/15682756/089202b8-2714-11e6-8558-282925d4d243.png)

I ran into an issue because of this today. I wanted to add a new page that would be in the subnavigation of a non-page menu item. Because there was no appropriate option for me to select a parent, the new page was considered a top-level item. And in my theme all top level items are automatically added to the menu at the top of the page. So when I published the page, it was added as a top-level item in my menu, which I definitely did not want. I had to then go into the menus area and delete that item from the menu, then re-add it where I wanted it.

So the upshot is that there is no way to do what I wanted to do without breaking my site for at least some amount of time. While this isn't a big deal for my personal site, this would be a big deal for a business site. Here is an example where I created a new page ("test page") without a parent, getting immediately added to the menu.

![should-not-be-in-menu](https://cloud.githubusercontent.com/assets/66797/15682769/165b6132-2714-11e6-8433-2e807a86faf1.png)

I'm not sure if this should be considered high priority or blocker. @designsimply what do you think?

cc @aduth

**Comment by codebykat** [comment]
I'm confused by this @rralian. Pages can't have non-page parents AFAIK (including in wp-admin).

Are you saying we should depart from core behavior and combine the "parent" selector in the editor with the menu settings? How would that even work under the hood?

**Comment by folletto** [comment]
Yeah, afaik it's default. I'd add that the whole flow happens as you mention if:

* The theme has a menu as "Default" — which in the latest analysis is a bit more than 85% — with no custom menu added *-OR-*
* Has a custom menu but with the `wp-admin` option of "Automatically add new top-level pages to this menu".

So it's likely to happen, but can be avoided if the site switched to a custom menu with that flag off.

It's still an interesting comment because highlights how the "site hierarchy" can be perceived in different ways. Often it's associated as the "Page hierarchy" but I can totally see how the Menu hierarchy might be referred to that too.

**Comment by rralian** [comment]
> Are you saying we should depart from core behavior and combine the "parent" selector in the editor with the menu settings? How would that even work under the hood?

@codebykat The "page parent" setting already dictates the menu hierarchy. If I publish a page without a parent, it automatically gets shown in the top menu. If I publish a page _with_ a page parent, it automatically gets placed into the menu as a subnav option underneath that parent. Check out my test site https://bobinprogress2.wordpress.com where I just published a new page (`Test Page 05/16`) with no parent, and then published a second page (`Test Page 2`) as a child of the first page. They both appear in the top menu in the way I expect, and I never touched any menu settings.

Because a menu can contain a non-page-item, I can put something _else_ in that top menu (like a link, or a post). Now if I want to add a page as a subnav option under that non-page, I cannot do it from the editor. I have no clear way to do this other than to publish the page and have it temporarily appear somewhere in that menu hierarchy in production until I can get to the menu screen to place it where I want it. But I just learned from the previous example that I _should_ be able to use this section to place my page in the menu.

As @folletto explains, this is situational and could potentially be avoided with a custom menu with a particular setting (the setting, btw is only available in wp-admin). But I think this is something we need to address better from the editor. Since "page parent" is already conflated with the menus, it would be most natural for me to be able to select _any menu item I want_ as the parent, and we just handle it accordingly behind the scenes, place it in the menu, and I suppose otherwise consider it a top-level page. However we do it, this component could really use some better explanation so the user knows what is going to happen (like an info popup).

**Comment by folletto** [comment]
> could potentially be avoided with a custom menu with a particular setting (the setting, btw is only available in wp-admin).

It's true that's available in `wp-admin` only, but just to be clear: we default it off so the behaviour doesn't present once a menu is saved once as custom — which is also annoying in the situation when one does want pages to be auto-added but at least avoids any issue.

> Since "page parent" is already conflated with the menus, it would be most natural for me to be able to select any menu item I want as the parent, and we just handle it accordingly behind the scenes, place it in the menu, and I suppose otherwise consider it a top-level page. 

While I agree on the principle behind this idea, I'm not sure about the solution... I fear it needs a more thorough rebuild of Pages and Menus as one thing, not just limited to the Editor. There are a lot of moving parts on Menus alone, so adding Pages and Editor UI to the mix makes it even more complicated and needs a review as one thing.

Brainstorming a bit...

1. Even if one might want to add a page as a sub-page of a non-page menu item, the URL hierarchy will still have to be strictly mapped to _something_, thus the page hierarchy can't really go away since it's mapped to URLs.
2. We could maybe break entirely the Menus / Pages auto-add. However, we know it works well for many users (the ones that use a simple site structure, as I guess it's largely the case), so I'm not sure. That would make any added page a two-steps process.
3. Maybe we can make the page hierarchy implicitly active in the slug/URL part of the editor (with a select parent dropdown there, might be tricky as UI) and have a menu editor on the side.
4. I wonder if seeing the page hierarchy in the Page view and then opening the editor with the Menu hierarchy in the sidebar won't be more confusing.
5. Converging Pages and Menus management as hierarchy view might make that selection easier in the Editor maybe? So the Editor doesn't manage the hierarchy ever, it's always in a different UI that can be switched easily to and a new page is always added at a specific location before even opening the Editor (ref: Menus).

_Aside: some services default pages as "hidden" (thus, published but not in the navigation) until they are explicitly added to the menu. That said, these services have FAR less complicated and far less flexible menus/pages structures._ 

tl;dr: this requires a more comprehensive design exploration.

**Comment by rralian** [comment]
> tl;dr: this requires a more comprehensive design exploration.

Agreed. This isn't something I expect to be quick fix. I just wanted to explain the issue, and that as a savvy user (debatable) I was still unable to publish a page and place it into the menu where I wanted without first breaking the navigation and having it appear in the top menu.

**Comment by lancewillett** [comment]
Removing from High priority since it's not a regression or blocking to main user flows in menus.
